tests/system/python/packages/test_north_azure.py

```diff
@@ -0,0 +1,601 @@
+__copyright__ = "Copyright (c) 2023 Dianomic Systems Inc"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+import subprocess
+import http.client
+import pytest
+import os
+import time
+import utils
+from pathlib import Path
+import urllib.parse
+import json
+import sys
+import datetime
+
+try:
+    subprocess.run(["python3 -m pip install azure-storage-blob==12.13.1"], shell=True, check=True)
+except subprocess.CalledProcessError:
+    assert False, "Failed to install azure-storage-blob modqule"
+
+from azure.storage.blob import BlobServiceClient
+
+# This  gives the path of directory where fledge is cloned. test_file < packages < python < system < tests < ROOT
+PROJECT_ROOT = subprocess.getoutput("git rev-parse --show-toplevel")
+SCRIPTS_DIR_ROOT = "{}/tests/system/python/scripts/package/".format(PROJECT_ROOT)
+SOUTH_SERVICE_NAME = "FOGL-7352_sysinfo"
+SOUTH_PLUGIN = "systeminfo"
+# ASSET = "FOGL-7352_system"
+NORTH_SERVICE_NAME = "FOGL-7352_azure"
+NORTH_PLUGIN_NAME = "azure-iot"
+NORTH_PLUGIN_DISCOVERY_NAME = "azure_iot"
+LOCALJSONFILE = "azure.json"
+FILTER = "expression"
+
+@pytest.fixture
+def reset_fledge(wait_time):
+    try:
+        subprocess.run(["cd {} && ./reset"
+                       .format(SCRIPTS_DIR_ROOT)], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "reset package script failed!"
+
+
+def read_data_from_azure_storage_container(azure_storage_account_url,azure_storage_account_key, azure_storage_container):
+    
+    try:
+        t1=time.time()
+        blob_service_client_instance = BlobServiceClient(account_url=azure_storage_account_url, credential=azure_storage_account_key)
+
+        container_client = blob_service_client_instance.get_container_client(container=azure_storage_container)
+
+        blob_list = container_client.list_blobs()
+
+        for blob in blob_list:
+            BLOBNAME = blob.name
+            print(f"Name: {blob.name}")
+
+
+        blob_client_instance = blob_service_client_instance.get_blob_client(azure_storage_container, BLOBNAME, snapshot=None)
+        with open(LOCALJSONFILE, "wb") as my_blob:
+            blob_data = blob_client_instance.download_blob()
+            blob_data.readinto(my_blob)
+        t2=time.time()
+        print(("It takes %s seconds to download "+BLOBNAME) % (t2 - t1))
+
+        # {iothub}    /{partition}/{YYYY}/{MM}/{DD}/{HH}/{mm}  
+        # foglamp-test/01        /2023  /04  /15  /20   /08.json
+        # LOCALFILE is the file path
+        # dataframe_blobdata = pd.read_csv(LOCALJSONFILE)
+        # print('the size of the data is: %d rows and  %d columns' % dataframe_blobdata.shape)
+        
+        with open(LOCALJSONFILE) as handler:
+            data = handler.readlines()
+            
+        print(data)
+        return data
+        
+    except (Exception) as ex:
+        print("Failed to read data due to {}".format(ex))
+        return None
+
+def verify_north_stats_on_invalid_config(fledge_url):
```

Comment:
Can't we reuse def verify_ping, both have similar functionality?

Reply:
With that we have to have to pass value of other arguments (skip_verify_north_interface, wait_time, retries) as well.